src/libsyntax_pos/lib.rs

```diff
@@ -83,16 +83,49 @@ pub struct MultiSpan {
 }
 
 impl Span {
+    #[inline]
+    pub fn new(lo: BytePos, hi: BytePos, ctxt: SyntaxContext) -> Self {
+        Span { lo, hi, ctxt }
+    }
+
+    #[inline]
+    pub fn lo(self) -> BytePos {
+        self.lo
+    }
+    #[inline]
+    #[must_use]
+    pub fn set_lo(self, lo: BytePos) -> Span {
+        Span::new(lo, self.hi(), self.ctxt())
```

Comment:
It was necessary with interning (the same is true for https://github.com/rust-lang/rust/pull/43968#discussion_r133887354). It will be necessary with any non-trivial `Span::new`/`Span::lo`/etc, so I left it.